corehq/apps/app_manager/models.py

```diff
@@ -2226,6 +2270,88 @@ def update_app_case_meta(self, app_case_meta):
                 meta.add_closer(self.unique_id, action.close_condition)
 
 
+class SchedulePhaseForm(IndexedSchema):
+    """
+    A reference to a form in a schedule phase.
+    """
+    form_id = FormIdProperty("modules[*].schedule_phases[*].forms[*].form_id")
+
+
+class SchedulePhase(IndexedSchema):
+    """
+    SchedulePhases are attached to a module.
+    A Schedule Phase is a grouping of forms that occur within a period and share an anchor
+    A module should not have more than one SchedulePhase with the same anchor
+
+    anchor:                     Case property containing a date after which this phase becomes active
+    forms: 			The forms that are to be filled out within this phase
+    """
+    anchor = StringProperty()
+    forms = SchemaListProperty(SchedulePhaseForm)
+
+    @property
+    def id(self):
+        """ A Schedule Phase is 1-indexed """
+        _id = super(SchedulePhase, self).id
+        return _id + 1
+
+    @property
+    def phase_id(self):
+        return "{}_{}".format(self.anchor, self.id)
+
+    def get_module(self):
+        return self._parent
+
+    _get_forms = IndexedSchema.Getter('forms')
+
+    def get_forms(self):
+        """Returns the actual form objects related to this phase"""
+        module = self.get_module()
+        return (module.get_form_by_unique_id(form.form_id) for form in self._get_forms())
+
+    def get_form(self, desired_form):
+        return next((form for form in self.get_forms() if form.unique_id == desired_form.unique_id), None)
+
+    def get_phase_form_index(self, form):
+        """
+        Returns the index of the form with respect to the phase
+
+        schedule_phase.forms = [a,b,c]
+        schedule_phase.get_phase_form_index(b)
+        => 1
+        schedule_phase.get_phase_form_index(c)
+        => 2
+        """
+        return next((phase_form.id for phase_form in self._get_forms() if phase_form.form_id == form.unique_id),
+                    None)
+
+    def remove_form(self, form):
+        """Remove a form from the phase"""
+        idx = self.get_phase_form_index(form)
+        if idx is None:
+            raise ScheduleError("That form doesn't exist in the phase")
+
+        self.forms.remove(self.forms[idx])
+
+    def add_form(self, form):
+        """Adds a form to this phase, removing it from other phases"""
+        old_phase = form.get_phase()
+        if old_phase is not None and old_phase.anchor != self.anchor:
+            old_phase.remove_form(form)
+
+        if self.get_form(form) is None:
+            self.forms.append(SchedulePhaseForm(form_id=form.unique_id))
+
+    def change_anchor(self, new_anchor):
+        if new_anchor is None or new_anchor.strip() == '':
+            raise ScheduleError(_("You can't create a phase without an anchor property"))
+
+        self.anchor = new_anchor
+
+        if self.get_module().phase_anchors.count(new_anchor) > 1:
```

Comment:
so if this happens, self has already had its anchor updated...is that going to be a problem?